### PR TITLE
Validate token and get service name

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

Use the existing validator in the library

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/services/auth/AuthService.java [hunk 1]
```diff
@@ -1,28 +1,22 @@
 package uk.gov.hmcts.reform.jobscheduler.services.auth;
 
-import feign.FeignException;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.validators.ServiceAuthTokenValidator;
 
 @Service
 public class AuthService {
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/services/auth/AuthService.java]: It looks like this class duplicates `ServiceAuthTokenValidator`. Maybe should be removed?

File: src/test/java/uk/gov/hmcts/reform/jobscheduler/FixtureData.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.jobscheduler;
+
+import feign.FeignException;
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.reform.authorisation.exceptions.AbstractAuthorisationException;
+
+import java.util.Collections;
+
+public final class FixtureData {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/jobscheduler/FixtureData.java:9]: is this different from existing `SampleData`?

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/jobscheduler/FixtureData.java:9]: kind of. at first i have put in `SampleData` but then figured - this is not actually a sample, it's more of a setup, a fixture of sorts. therefore a new class to not confuse with logical elements within one and another

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/services/auth/AuthService.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/jobscheduler/services/auth/AuthService.java]: true. did not see prior removing all this. will get on this next thing

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/Application.java [hunk 3]
```diff
@@ -3,11 +3,9 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.cloud.client.circuitbreaker.EnableCircuitBreaker;
-import org.springframework.cloud.netflix.feign.EnableFeignClients;
 import org.springframework.cloud.netflix.hystrix.dashboard.EnableHystrixDashboard;
 
 @SpringBootApplication
-@EnableFeignClients
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/Application.java:10]: can we also remove the `spring-cloud-starter-openfeign` dependency from build.gradle?

File: src/test/java/uk/gov/hmcts/reform/jobscheduler/FixtureData.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    private FixtureData() {
+        // empty constructor
+    }
+
+    public static AbstractAuthorisationException setUpUnauthorisedAccess() {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/jobscheduler/FixtureData.java]: The method name kind of reads weird (especially when used: _'given XXX will Throw setUpUnauthorisedAccess'_.
It's not taking any parameters, nor is it modifying any state - it's creating an exception.

Perhaps `getAuthorisationException`?

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/Application.java (hunk 3 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/jobscheduler/Application.java:10]: all libs come from `service-auth-provider-client` - should be good to remove. planned to do so too

**Review by KarolJastrzebski — APPROVED** [review]

**Pull request merged**